src/components/transactions/TransactionItem.vue

```diff
@@ -1,24 +1,63 @@
 <template>
-  <div class="tx">
-    <TransactionIcon :transaction-group="transaction.group"></TransactionIcon>
-    <TransactionDetails
-      :show-meta-data="showMetaData"
-      :transaction="transaction"
-      :validators="validators"
-      :address="address"
-    />
+  <div class="tx-container">
+    <div class="tx" @click="toggleDetail">
+      <component
+        :is="messageTypeComponent"
+        :transaction="transaction"
+        :validators="validators"
+        :session-address="address"
+      />
+      <div class="toggle" :class="{ up: show }">
+        <i class="material-icons toggle-icon">keyboard_arrow_down</i>
+      </div>
+    </div>
+    <transition name="slide-out">
+      <div v-if="show" class="tx-details">
+        <TransactionMetadata v-if="showMetaData" :transaction="transaction" />
+      </div>
+    </transition>
   </div>
 </template>
 
 <script>
-import TransactionIcon from "./TransactionIcon"
-import TransactionDetails from "./TransactionDetails"
+import { messageType } from "./messageTypes.js"
```

Comment:
here is the messageType enum I would have liked to see in the icon handling as well. no need for a separate form of declaring the types.

Reply:
But we can do that as a follow up.